cost-analyzer/templates/cost-analyzer-deployment-template.yaml

```
@@ -21,7 +21,7 @@ metadata:
   {{- end }}
 spec:
 {{- if .Values.kubecostDeployment }}
-  replicas: {{ .Values.kubecostDeployment.replicas | default 1 }}
+  replicas: {{ toString .Values.kubecostDeployment.replicas | default 1 | int }}
```

```suggestion
  replicas: {{ toString .Values.kubecostDeployment.replicas | int }}
```

One minor change if you're open to it. From a UX perspective, we try to have all defaults defined in the `values.yaml` as opposed to hidden in the templates ([ref](https://github.com/kubecost/cost-analyzer-helm-chart/blob/develop/DEVELOPMENT.md#yaml-style-guidelines)). Let me know what you think!

---

That works for me!

If that's the case....could we just drop both `toString` and `int`, and have `{{ .Values.kubecostDeployment.replicas }}` ?

---

Hm, I _think_ we would still need the `toString` and `int`.

```yaml
# values.yaml
kubecostDeployment:
  replicas: 1  # default value of 1, but what if somebody sets this to 0?
```

---

If somebody sets `0` in their `values.yaml`, but we aren't passing the value through the `default` function, then it will just get translated directly into the template, right?

Like, we stringify the value so that it isn't "falsey" when it runs through the `default` function, and then we go back to an int. But, if there's no `default` function to worry about, then I don't think we need to do the `int -> string -> int` dance.

I'll just try it out and post results - probably better than any amount of guessing :)

---

Tried it out! It works. If you want to verify, you can pull down the updated branch and with values.yaml of

```
kubecostDeployment:
  replicas: 0
```
Run the install and see that we get no cost-analyzer pods!

---

Wooh thanks for testing!